be/src/io/fs/file_system.cpp

```diff
@@ -44,6 +46,24 @@ Status FileSystem::delete_file(const Path& file) {
     FILESYSTEM_M(delete_file_impl(path));
 }
 
+Status FileSystem::delete_directory_or_file(const Path& path) {
+    auto real_path = absolute_path(path);
```

Comment:
I think we should implement this only in `LocalFileSystem`.
Because you use `stat` to check if it is dir, which is only available for local fs.

And there is already a method `is_directory()` in local fs, better to use that.